File: JavaSource/src/main/java/ru/akhcheck/patterns/state/states/Green.java
package ru.akhcheck.patterns.state.states;

import ru.akhcheck.patterns.state.Direction;
import ru.akhcheck.patterns.state.TrafficLight;

/** Зелёный свет: следующим всегда идёт жёлтый в сторону красного. */
public class Green implements State {
    @Override
    public void switchState(TrafficLight light) {
        light.setState(new Yellow());
        light.setDirection(Direction.TO_RED);
    }

    @Override
    public String getStringState() {
        return "Green";
    }
}

File: State/python-source/states/Green.py
from __future__ import annotations

from typing import TYPE_CHECKING

from .State import State

if TYPE_CHECKING:
    from TrafficLight import TrafficLight


class Green(State):
    def switch(self, light: TrafficLight) -> None:
        from .Yellow import Yellow

        light.set_state(Yellow())
        light.set_direction(light.TO_RED)

    def get_string_state(self) -> str:
        return "Green"
